rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -0,0 +1,365 @@
+                )
+                liquidity_pool = LiquidityPool(
+                    asset=lp_uniswap_v2_asset,
+                    assets=liquidity_pool_assets,
+                    total_supply=lp_total_supply,
+                    user_balance=Balance(amount=user_lp_balance),
+                )
+                address_balances[user_address].append(liquidity_pool)
+
+            # Update pagination step
+            param_values['offset'] += limit  # type: ignore
+
+        protocol_balance = ProtocolBalance(
+            address_balances=dict(address_balances),
+            known_assets=known_assets,
+            unknown_assets=unknown_assets,
+        )
+        return protocol_balance
+
+    def _get_balances_zerion(
+        self,
+        addresses: List[ChecksumEthAddress],
+    ) -> ProtocolBalance:
+        """Get the addresses' pools data via Zerion SDK.
+        """
+        address_balances: DDAddressBalances = defaultdict(list)
+        known_assets: Set[EthereumToken] = set()
+        unknown_assets: Set[UnknownEthereumToken] = set()
+
+        protocol_balance = ProtocolBalance(
+            address_balances=dict(address_balances),
+            known_assets=known_assets,
+            unknown_assets=unknown_assets,
+        )
+        return protocol_balance
+
+    def _get_known_asset_price(
+        self,
+        known_assets: Set[EthereumToken],
+        unknown_assets: Set[UnknownEthereumToken],
+    ) -> AssetPrice:
+        """Get the tokens prices via Inquirer
+
+        Given an asset, if `find_usd_price()` returns zero, it will be added
+        into `unknown_assets`.
+        """
+        asset_price: AssetPrice = {}
+
+        for known_asset in known_assets:
+            asset_usd_price = Inquirer().find_usd_price(known_asset)
+
+            if asset_usd_price != Price(ZERO):
+                asset_price[known_asset.ethereum_address] = FVal(asset_usd_price)
+            else:
+                unknown_asset = UnknownEthereumToken(
+                    identifier=known_asset.identifier,
+                    ethereum_address=known_asset.ethereum_address,
+                    name=known_asset.name,
+                    decimals=known_asset.decimals,
+                )
+                unknown_assets.add(unknown_asset)
+
+        return asset_price
+
+    def _get_unknown_asset_price_graph(
+            self,
+            unknown_assets: Set[UnknownEthereumToken],
+    ) -> AssetPrice:
+        """Get today's tokens prices via the Uniswap subgraph
+
+        Uniswap provides a token price every day at 00:00:00 UTC
+        """
+        asset_price: AssetPrice = {}
+
+        unknown_assets_addresses = [asset.ethereum_address for asset in unknown_assets]
+        unknown_assets_addresses_lower = (
+            self._convert_addresses_to_lower_case(unknown_assets_addresses)
+        )
+
+        querystr = format_query_indentation(TOKEN_DAY_DATAS_QUERY.format())
+        limit = 1000
```

Comment:
Funny thing because I thought I had read somewhere the subgraphs are hard limited to 1000 entries per request, but I can't find it now. They mention the pagination [here](https://thegraph.com/docs/graphql-api#queries#Pagination), though.

From my short experience, Balancer and [AAVE](https://docs.aave.com/developers/integrating-aave/using-graphql#common-gotchas) subgraphs are limited.

Despite it is improbable to have more than 1000 LPs and 1000 unknown tokens, I've introduced the pagination for starting to be aware about the matter; as for historical trades it is a must.

At the moment it is doing an unnecessary extra request that can be removed if I check if the return length is less than the limit.